gp/genome.py

```diff
@@ -257,15 +273,17 @@ def _validate_dna(self, dna):
                     "be identical to non-coding allele"
                 )
 
-    def _hidden_column_idx(self, region_idx):
+    def _hidden_column_idx(self, region_idx: int) -> int:
         assert self._n_inputs <= region_idx
         assert region_idx < self._n_inputs + self._n_rows * self._n_columns
         hidden_column_idx = (region_idx - self._n_inputs) // self._n_rows
         assert 0 <= hidden_column_idx
         assert hidden_column_idx < self._n_columns
         return hidden_column_idx
 
-    def iter_input_regions(self, dna=None):
+    def iter_input_regions(
+        self, dna: Optional[List[int]] = None
+    ) -> Generator[Tuple[int, list], None, None]:
```

Comment:
what do the last `None, None` indicate?

Reply:
The `SendType` and `ReturnType`. See https://docs.python.org/3/library/typing.html#typing.Generator

Reply:
thanks!